frontend/src/components/WaitingRoom/PrecallNetworkTestDialog/PrecallNetworkTestDialog.tsx

```diff
@@ -0,0 +1,372 @@
+      <DialogTitle sx={{ m: 0, p: 0, pl: 3, pt: 3, backgroundColor: theme.colors.surface }}>
+        <Typography
+          component="div"
+          variant="h5"
+          sx={{
+            fontWeight: theme.typography.weight['body-base'].value,
+            color: theme.colors.textSecondary,
+          }}
+        >
+          {t('waitingRoom.precallNetworkTest.title')}
+        </Typography>
+        <IconButton
+          aria-label={t('button.close')}
+          onClick={handleClose}
+          sx={{
+            position: 'absolute',
+            right: 16,
+            top: 16,
+            color: theme.colors.secondary,
+          }}
+        >
+          <VividIcon name="close-line" customSize={-5} />
+        </IconButton>
+      </DialogTitle>
+      <DialogContent sx={{ backgroundColor: theme.colors.surface, pt: 2 }}>
+        <Box sx={{ display: 'flex', flexDirection: 'column', gap: 3, py: 2 }}>
+          <Typography
+            variant="body1"
+            sx={{
+              color: theme.colors.textSecondary,
+              fontWeight: theme.typography.weight['body-base'].value,
+              textAlign: 'left',
+            }}
+          >
+            {t('waitingRoom.precallNetworkTest.subtitle')}
+          </Typography>
+
+          {state.isTestingQuality && (
+            <Box sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}>
+              <Box
+                sx={{
+                  display: 'flex',
+                  flexDirection: 'column',
+                  alignItems: 'center',
+                  gap: 3,
+                  py: 4,
+                  backgroundColor: theme.colors.background,
+                }}
+              >
+                <CircularProgress size={60} />
+              </Box>
+              <Button
+                variant="outlined"
+                onClick={handleStopTest}
+                sx={{
+                  mt: 2,
+                  backgroundColor: theme.colors.primary,
+                  color: theme.colors.onPrimary,
+                  minWidth: '400px',
+                  display: 'flex',
+                  gap: 2,
+                }}
+              >
+                <VividIcon name="stop-line" customSize={-6} />
+                {t('waitingRoom.precallNetworkTest.stopTest')}
+              </Button>
+            </Box>
+          )}
+
+          {!state.isTestingQuality && (state.qualityResults || state.error) && (
+            <Box sx={{ display: 'flex', flexDirection: 'column', gap: 3 }}>
+              {state.error ? (
+                <Box
+                  sx={{
+                    display: 'flex',
+                    flexDirection: 'column',
+                    alignItems: 'center',
+                    gap: 2,
+                    py: 2,
+                  }}
+                >
+                  <Typography variant="h6" sx={{ color: theme.colors.error, textAlign: 'center' }}>
```

Comment:
<img width="388" height="180" alt="Image" src="https://github.com/user-attachments/assets/671bf232-5e2f-4908-9ddd-41f37b2deb8e" />

Consider adding flex layout properties to the error Typography to properly align the icon and text:

```suggestion
                     sx={{
                      color: theme.colors.error,
                      textAlign: 'center',
                      display: 'flex',
                      alignItems: 'center',
                      gap: 2,
                    }}
```